api/helpers/ find-spectatable-games.js

```diff
@@ -0,0 +1,23 @@
+const dayjs = require('dayjs');
+
+module.exports = {
+  friendlyName: 'Find Spectatable Games',
+
+  description: 'Finds all the games that are available to spectate',
+
+  fn: async (_, exits) => {
+
+    const recentUpdateThreshhold = dayjs().subtract(10, 'minute').valueOf();
+
+    try {
+      const games = await Game.find({
+        status: false, // not open to join
+        result: -1, // not finished yet
```

Comment:
This is a weird key name for this, but it's probably based on the DB so I don't think it matters as much. It's a little confusing to me though.

Reply:
That smells; I'd love to pick your brain on how to clarify it. 

So the idea here is to find games that are being actively played. To do so we're querying for games with two criteria:

1. `status: false` -- `status` is a boolean for whether a game is open for players to join. It defaults to true when the game is made and is set to `false` when a 2nd player joins. So `false` means the game has at least two people in it. 

2. `result: -1` -- `result` is an enum describing the end state of a game. -1 is the initial value before a game starts and then when a game ends it's set to 0 if p0 wins, 1 if p1 wins, and 2 for a stalemate. 

In the short term at minimum this should be compared using an enum value (Result.INCOMPLETE) instead of the raw -1. More importantly, I believe there is actually a bug here where games that have two players (so status = false) but that haven't started yet (not everyone is ready) would be turned up by this, possibly letting you spectate a game that hasn't been dealt yet. I'll see if I can replicate in a test case; it should be fixable by checking p0Ready and p1Ready, but that is a bit of a kludge and speaks to the awkwardness of the data architecture here.

In the longer term, I think this hits on the smell of having these two separate fields (result and status) for what is logically more one state on a spectrum of values, ie a game is created, started, and  ends with one of three possible states (p0 win, p1 win, draw). These two columns (status and result) could likely be collapsed into one column that represents which of those states the game is in. It would be important to name it clearly and it would potentially have downsides in that some checks would need to compare this new column to a list of values instead of just one. That point could use further baking, penny for your thoughts. If we can land on a coherent vision for it, I'll write up an issue

Reply:
I really like the idea of combining this in to a single call because at the end of the day they’re all just games. We only care about properties on them that say whether they’re observable, playable, etc.

Reply:
So another way this could go down would be to have two columns:
`status` which would be an enum and `winner` which would be the id of the user who won.

`status` could allow the values:
`UNSTARTED`, `STARTED`, `FINISHED`

So games that are `game.status === Status.FINISHED` that have `game.winner === null` would be stalemates, and otherwise, the `winner` field would tell you which player won. 

What this doesn't properly capture in and of itself is that the `status` variable currently represents whether the game is joinable. That may or may not be worth capturing separately. In principle you could omit this from the database and use the count of players to determine this i.e. you can't join unless there's < 2 players already in the game. The thing that this sorta misses is the ability to manually close games in the database by flipping their status to false, which I sometimes do to clean them up. There's a way to get all of these, I'm sure, but I don't have it fully mapped out yet.

@seriouslysean this speaks to a documentation question: given that we know this is something that we're interested to refactor but that we don't currently have a clear and specific vision for, should we write up an issue for it? How should we avoid losing the seed of this idea while acknowledging that it's underbaked for execution? Could be an issue with a needs-refinement label perhaps

Reply:
Test case and improvement is in place to fix the bug around joining unstarted games. And https://github.com/cuttle-cards/cuttle/issues/374 is documented to investigate a reorganization of these database columns